File: src/main/java/github/lms/approval/application/ApproveCommand.java
package github.lms.approval.application;

import java.util.UUID;

public record ApproveCommand(
        UUID approvalId,
        UUID stepId,
        UUID approverId,
        String idempotencyKey
) {
    public ApproveCommand {
        if (approvalId == null) {
            throw new IllegalArgumentException("approvalId is required");
        }
        if (stepId == null) {
            throw new IllegalArgumentException("stepId is required");
        }
        if (approverId == null) {
            throw new IllegalArgumentException("approverId is required");
        }
        if (idempotencyKey == null || idempotencyKey.isBlank()) {
            throw new IllegalArgumentException("idempotencyKey is required");
        }
    }
}

File: src/main/kotlin/github/lms/approval/application/ApproveCommand.kt
package github.lms.approval.application

import java.util.*

data class ApproveCommand(
    val approvalId: UUID,
    val stepId: UUID,
    val approverId: UUID,
    val idempotencyKey: String
) {
    init {
        require(idempotencyKey.isNotBlank()) { "idempotencyKey is required" }
    }
}
